tests/test_mouse.py

```diff
@@ -0,0 +1,193 @@
+import platform
+from collections import namedtuple
+
+from syncer import sync
+
+from tests.conftest import chrome_only
+
+
+dimensions = """() =>
+    {
+      const rect = document.querySelector('textarea').getBoundingClientRect();
+      return {
+        x: rect.left,
+        y: rect.top,
+        width: rect.width,
+        height: rect.height
+      };
+    }
+"""
+
+
+async def evaluate_dimensions(isolated_page):
+    """Execute `dimensions` script and parse value to namedTuple with x, y, width and height fields."""
+    result = await isolated_page.evaluate(dimensions)
+    rectTuple = namedtuple("rect", "x y width height")
+    return rectTuple(x=result['x'], y=result['y'], width=result['width'], height=result['height'])
+
+
+@sync
+async def test_click_document(isolated_page):
+    """should click the document"""
+    await isolated_page.evaluate("""
+    () => {
+      window.clickPromise = new Promise(resolve => {
+        document.addEventListener('click', event => {
+          resolve({
+            type: event.type,
+            detail: event.detail,
+            clientX: event.clientX,
+            clientY: event.clientY,
+            isTrusted: event.isTrusted,
+            button: event.button
+          });
+        });
+      });
+    }
+    """)
+    await isolated_page.mouse.click(50, 60)
+    event = await isolated_page.evaluate("window.clickPromise")
+    assert event['type'] == 'click'
+    assert event['detail'] == 1
+    assert event['clientX'] == 50
+    assert event['clientY'] == 60
+    assert event['isTrusted']
+    assert event['button'] == 0
+
+
+@sync
+@chrome_only
+async def test_resize_textarea(isolated_page, server):
+    """should resize the textarea"""
+    page = isolated_page
+    await page.goto(server / 'input/textarea.html')
+    x, y, width, height = await evaluate_dimensions(page)
+    mouse = page.mouse
+    await mouse.move(x + width - 4, y + height - 4)
+    await mouse.down()
+    await mouse.move(x + width + 100, y + height + 100)
+    await mouse.up()
+    newDimensions = await evaluate_dimensions(page)
+    assert newDimensions.width == round(width + 104)
+    assert newDimensions.height == round(height + 104)
+
+
+@sync
+@chrome_only
+async def test_select_text_with_mouse(isolated_page, server):
+    """should select the text with mouse"""
+    page = isolated_page
+    await page.goto(server / 'input/textarea.html')
+    await page.focus('textarea')
+    text = "This is the text that we are going to try to select. Let\'s see how it goes."
+    await page.keyboard.type(text)
+    # Firefox needs an extra frame here after typing or it will fail to set the scrollTop
+    await page.evaluate("() => new Promise(requestAnimationFrame)")
```

Comment:
When it comes to these single line anonymous function, it seems you can just use the statement directly (even though it is that way in the puppeteer source). This goes for every other anonymous function as well.
```suggestion
    await page.evaluate("new Promise(requestAnimationFrame)")
```

Reply:
Got rid of unnecessary anonymous funcs.